### ISSUE TITLE
Mouse-over window in EssentialsX Chat (timestamps)

### ISSUE THREAD

**Issue by takatalvi** [EssentialsX/Essentials#2655]
Hi, I've been wondering about the possibility of having timestamps in chat messages. While the chat format is modifiable and I think it also supports time, I always thing that less is more and a timestamp in every single chat line would kind of flood the whole chat. Instead, the **timestamp could be seen in a mouse-over box.** In vanilla chat, messages are interactive - you can click on a player name and you will start a private message with them. You can hover over an entity name or item name in death messages and you would see their data. In the same manner, timestamps could be displayed over playername in EssentialsX Chat messages. Maybe this would allow for even more info, this could be configurable using placeholders in the Chat section of the EssX config.

**Comment by mdcfe** [comment]
Related: #2391

Currently, EssentialsX Chat uses Bukkit's standard chat formatting methods. This means we're limited in what we can do to chat messages before they get sent out to players, and as a result we can't access or modify the JSON chat components that get sent to the player.

In addition, EssentialsX doesn't currently support chat components. I'd like to implement this soon using [KyoriPowered/text](https://github.com/KyoriPowered/text), but EssentialsX Chat could be updated to support components without waiting on EssentialsX.

**Comment by takatalvi** [comment]
Too bad it's limited by Bukkit but if it's possible to go around it, that would be a very nice addition :)

**Comment by mdcfe** [comment]
We could work around it by cancelling the Bukkit chat event and sending our own message, but this isn't ideal and might break other plugins.

Another potential alternative is to look into whether Spigot or Paper have extended API for this, but I don't know if this is the case.

**Comment by pop4959** [comment]
As MD mentioned -- this kind of change will have to be postponed until global chat changes are made. Please feel free to follow the discussion issue linked, and post your feedback for features that should be considered such as this one.
